### PR TITLE
Cleaner org transfer

### PR DESCRIPTION
**Pull request by ryanpbrewster**
@camhashemi 

Minor assorted cleanup

### CODE REVIEW

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/OrganizationMembershipCommander.scala [hunk 1]
```diff
@@ -241,19 +252,23 @@ class OrganizationMembershipCommanderImpl @Inject() (
...
-        (membership.role, newMembership.role) match { // assumes admins can only be added via modifying a membership, not creating one
-          case (OrganizationRole.MEMBER, OrganizationRole.ADMIN) => planCommander.registerNewAdmin(org.id.get, request.targetId, ActionAttribution(user = Some(request.requesterId), admin = None))
-          case (OrganizationRole.ADMIN, OrganizationRole.MEMBER) => planCommander.registerRemovedAdmin(org.id.get, request.targetId, ActionAttribution(user = Some(request.requesterId), admin = None))
-          case _ =>
+        if (membership.role == request.newRole) Right(OrganizationMembershipModifyResponse(request, membership))
+        else {
+          val org = orgRepo.get(request.orgId)
+          val newMembership = orgMembershipRepo.save(membership.withRole(request.newRole))
+          (membership.role, newMembership.role) match {
+            // assumes admins can only be added via modifying a membership, not creating one
```
Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/OrganizationMembershipCommander.scala]: that seems bad :(

Comment (camhashemi) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/OrganizationMembershipCommander.scala]: you can remove this comment if you add a check for `request.newRole == ADMIN` in `unsafeAddMembership`, where if true you call `planCommander.registerNewAdmin`. I can do it as well, though seems you're in here.

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/OrganizationCommander.scala [hunk 2]
```diff
@@ -418,29 +418,25 @@ class OrganizationCommanderImpl @Inject() (
...
-      case Some(orgFail) => Left(orgFail)
-      case None =>
-        db.readWrite { implicit session =>
+    db.readWrite { implicit session =>
+      getValidationError(request) match {
+        case Some(orgFail) => Left(orgFail)
+        case None =>
           val org = orgRepo.get(request.orgId)
-          orgMembershipRepo.getByOrgIdAndUserId(org.id.get, request.newOwner, excludeState = None) match {
-            case Some(membership) if membership.isActive =>
```
Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/OrganizationCommander.scala:428]: We actually check for this in `getValidationError`. You aren't allowed to add a member that is already in the org.

Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/OrganizationCommander.scala:428]: At some point it'd be interesting to look at merging add/modify into a single method, `internMembership`. Simpler to use. Simpler to reason about.

**Pull request merged**